stl/inc/algorithm

```diff
@@ -2834,6 +2860,23 @@ namespace ranges {
             _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se, _It>);
             _STL_INTERNAL_STATIC_ASSERT(indirect_binary_predicate<ranges::equal_to, projected<_It, _Pj>, const _Ty*>);
 
+#if _USE_STD_VECTOR_ALGORITHMS
+            if constexpr (is_same_v<_Pj, identity> && _Vector_alg_in_find_is_safe<_It, _Ty>
+                          && sized_sentinel_for<_Se, _It>) {
+                if (!_STD is_constant_evaluated()) {
+                    const auto _First_ptr = _To_address(_First);
+                    const auto _Last_ptr  = _First_ptr + (_Last - _First);
+
+                    const auto _Result = __std_find_last_trivial(_First_ptr, _Last_ptr, _Value);
+                    if constexpr (is_pointer_v<_It>) {
+                        return {_Result, _STD move(_Last)};
+                    } else {
+                        return {_STD move(_First) + (_Result - _First_ptr), _STD move(_Last)};
+                    }

```

Comment:
This is better, but `_STD move(_Last)` is attempting to return `_Se _Last` as the end of a `subrange<_It>`. We know that `sized_sentinel_for<_Se, _It>` holds here, but in general we can't convert `_Se` to `_It`.

@CaseyCarter self-assigned this PR and presumably knows how to fix this (again I could try and probably get something to work).